### PR TITLE
Fix: pre-release versions are not considered when detecting outdated dependencies.

### LINKED ISSUE #572: Don't penalise for not supporting latest dev dependencies
My plugin depends on the latest stable release of rxdart (0.22.6) but is penalised -10 points because 0.23.0-dev.1 is available.

### PR DESCRIPTION
**Pull request by isoos**
- fixes #572 
- end2end test confirmed fix on [angular_components 0.10.0](https://pub.dev/packages/angular_components/versions/0.10.0#-analysis-tab-)

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -622,7 +622,7 @@ class PkgDependency implements Comparable<PkgDependency> {
 
   bool get isLatest => available == null;
 
-  bool get isOutdated => !isLatest;
+  bool get isOutdated => available != null && !available.isPreRelease;
```
Comment (sigurdm) [inline lib/src/model.dart]: Can it happen that there is both a newer prerelease and a  newer nonprerelease available, and the prerelease shadows the non-prerelease making the dependency seem not outdated even though it is?

Comment (isoos) [reply lib/src/model.dart]: The information is coming from the `pub` client, and we'll take here whatever it provides.

Any memory what `pub` uses? (latest stable, latest by date, or just highest version value?) I don't see much risk of using this at face value right now, but later on we could also query it from  from pub.dev.

Comment (sigurdm) [reply lib/src/model.dart]: True, at worst we will warn too little.

I don't know off hand what the pub client does.
Perhaps add a TODO that we might want to investigate edge cases here.

**Comment by isoos** [comment]
CI is green after rebase, but did not update the status here. Merging.

**Pull request merged**